### PR TITLE
Add support for setting priority for classifier programs

### PR DESCRIPTION
**Pull request by anfredette**
Addresses issue #358

The SchedClassifier attach function does not currently allow the priority (aka preference) of a classifier program to be explicitly set. Instead, the underlying netlink_qdisc_attach function defaults the priority to zero which tells the system to assign it the next highest priority for the given qdisc. It is useful to be able to explicitly control the order in which filters are executed and also to be able to insert new filters into an existing set at less than the highest priority.

This change adds support for setting the priority explicitly.

Note: aya-template will also need to be updated to add the new parameter.

Signed-off-by: Andre Fredette <afredette@redhat.com>

### CODE REVIEW

File: aya/src/programs/tc.rs [hunk 1]
```diff
@@ -120,12 +124,13 @@ impl SchedClassifier {
         &mut self,
         interface: &str,
         attach_type: TcAttachType,
+        priority: u32,
```
Comment (alessandrod) [inline aya/src/programs/tc.rs:127]: I think this should be `u16` in the API if the valid range is `0..u16::MAX`, then we should cast internally.

**Review by alessandrod — COMMENTED** [review]
Thank you!

See minor nit and failing tests.

File: aya/src/programs/tc.rs (hunk 1 above)
Comment (anfredette) [reply aya/src/programs/tc.rs:127]: @alessandrod thanks for the review.  I went back and forth on this myself.  Using a u16 requires more code changes and conversions back and forth, so I ended up keeping it as a u32 in the API.  However, making it a u16 is probably best for the user, so I'll make that change.

**Comment by alessandrod** [comment]
Self reminder: this is a breaking change so requires a version bump

File: aya/src/programs/tc.rs (hunk 1 above)
Comment (anfredette) [reply aya/src/programs/tc.rs]: The question I had was whether to just change priority to u16 in the SchedClassifier attach API, or everywhere in tc.rs and netlink.rs.  I chose everywhere to keep it consistent.  What do you think?

Comment (anfredette) [reply aya/src/programs/tc.rs]: FWIW, the iproute2 code has priority/preference as a u32 and does a range check:
https://github.com/shemminger/iproute2/blob/main/tc/tc_filter.c#L150

Comment (alessandrod) [reply aya/src/programs/tc.rs]: Do you know if there's a rationale for this? Is the range expected to be increased?

Comment (anfredette) [reply aya/src/programs/tc.rs]: Not that I know of.  They also do a <<16 [here](https://github.com/shemminger/iproute2/blob/main/tc/tc_filter.c#L186) like we're doing, so only the lower 16 bits are used anyway.  I think it's more user-friendly for us to make it a u16 at least in the API.

Comment (alessandrod) [reply aya/src/programs/tc.rs]: Yep agreed!

**Review by alessandrod — APPROVED** [review]

**Pull request merged**